### ISSUE TITLE
Idea: Modernize code (use typescript, use class syntax instead of prototype)

### ISSUE THREAD

**Issue by TysonAndre** [parse-community/node-apn#68]
A switch to typescript may make sense in a major version

index.d.ts may get out of sync with the implementation. It may be easier to write everything in typescript and add a publish hook to convert code to typescript
(https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-dynamodb/package.json https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-dynamodb/tsconfig.json)

(the publish hook would trigger a typescript build to dist/ or a subdirectory)

Alternately, with --checkJs, jsdoc can be type checked in CI

----

Also, class syntax can be used in node 6+ - the upstream code was supporting node 4+ and declared `function Client` instead of `class Client` https://github.com/parse-community/node-apn/blob/master/lib/client.js

**Comment by davimacedo** [comment]
Yes. I think that it makes sense and I'd be glad to review a PR.

**Comment by mtrezza** [comment]
Added a bounty so hopefully someone will pick this up.

**Comment by mrevanzak** [comment]
is this still valid?

**Comment by mtrezza** [comment]
@mrevanzak yes, the issue is still up for grabs

**Comment by mrevanzak** [comment]
I would give it a try if it still $100

**Comment by paulovitin** [comment]
@mtrezza. is this still valid?
